### PR TITLE
Adds minimap drawing

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
A person using the CIC drawing tablet can now draw on the map

Requires SL+ level of leadership skill, being banned from ANY command role restricts you from using it

While funny dont use this exclusively for drawing phallic icons or it will be removed

How to use:
You get 4 colors + an eraser, clear all and labeller
Click on the tool to select it right click to unselect it
Colors and eraser are used by right clicking to place one pixel or drag-release to draw a line between the start and end of the drag
Clear all removes all lines
Labeller lets you place labels on the map with up to 35 character size
Right click labeller to remove all labels
## Why It's Good For The Game

Something something coordination

## Changelog
:cl:
add: Added minimap drawing tablets to CIC
imageadd: "erase" and "erase all" sprites by Orcacora
admin: added the minimap drawing logging to game log: MINIMAP_DRAW: text
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
finally
this but you dont have to make an imgur link and link it on accnounments

**Comment by chizzy376** [comment]
How do you erase drawn lines? Or are they temporary and only last a minute or two before fading?

**Comment by TiviPlus** [comment]
Permanent, you get undo last line + clear all lines + drawing with an eraser
Ill add usage instructions to pr body

File: code/game/objects/items/devices/minimap_tablet.dm [hunk 1]
```diff
@@ -0,0 +1,304 @@
+///A player needs to be unbanned from ALL these roles in order to be able to use the minimap drawing tool
+GLOBAL_LIST_INIT(roles_allowed_minimap_draw, list(SQUAD_LEADER, CAPTAIN, FIELD_COMMANDER, STAFF_OFFICER))
```
Comment (Lumipharon) [inline code/game/objects/items/devices/minimap_tablet.dm:2]: The ability to draw like a 2 year old is now the most protected feature in the game. Truly befitting the unga.

**Review by Lumipharon — APPROVED** [review]

**Comment by Losenis** [comment]
Please add a freedrawing option, it'd be great to have curved arrows and the sort

**Comment by novaepee** [comment]
A NEW AGE OF TGMC IS HERE! WITNESS ITS GLORY!

**Comment by blackdav123** [comment]
queens could definitely use this too. would make it much easier to manage who pushes where and where boilers and crushers need to go

**Comment by XSlayer300** [comment]
do we have set rules for the minimap though?

**Comment by novaepee** [comment]
new feature

Great feature all around.

File: code/game/objects/items/devices/minimap_tablet.dm [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+	active_mouse_icon = 'icons/UI_Icons/minimap_mouse/label.dmi'
+	screen_loc = "16,8"
+	/// List of turfs that have labels attached to them. kept around so it can be cleared
+	var/list/turf/labelled_turfs = list()
+
+/atom/movable/screen/minimap_tool/label/Click(location, control, params)
+	. = ..()
+	var/list/modifiers = params2list(params)
+	if(modifiers[BUTTON] == RIGHT_CLICK)
+		log_minimap_drawing("[key_name(usr)] has cleared current labels")
```
Comment (Autisem) [inline code/game/objects/items/devices/minimap_tablet.dm:258]: could this be made to work so that if an aghosted admin does this they can delete it too?

**Comment by chizzy376** [comment]
Please allow us to delete text and make a [YES / NO] confirmation box for wiping all the chalk off the board

**Comment by FosterGrime-Heymynameslouie** [comment]
this is so sad

File: code/game/objects/items/devices/minimap_tablet.dm (hunk 2 above)
Comment (TiviPlus) [reply code/game/objects/items/devices/minimap_tablet.dm:258]: could yes but I dont feel like doing it in this pr

**Comment by FosterGrime-Heymynameslouie** [comment]
we are so back

**Comment by AlexCrimson** [comment]
Could this be disabled for ghost/observer? Or at least a toggle defaulted to off? Seeing the marines plan right there on the map before people spawn as a xeno is kinda meta. 

Maybe not a big deal, as people can just... not meta.

**Comment by TiviPlus** [comment]
Not feasible with current minimap code

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**